src/model/block/IDirectorySelectionBlockConfig.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Flow Interoperability Project (flowinterop.org)
+ * Flow Runner
+ * Copyright (c) 2019, 2020 Viamo Inc.
+ * Authored by: Brett Zabos (brett.zabos@viamo.io)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ **/
+
+import {IPromptConfig} from '../..'
+
+export interface IDirectorySelectionBlockConfig extends IPromptConfig<string | null> {
+  kind: string
+  prompt: string
+  promptAudio: string
+  primaryField: string
+  secondaryFields: [string]
+  choiceRowFields: [string]
+  choiceRows: [[string]]
+  responseFields: [string]
+}
```

Comment:
If I do this check: `this.props.prompt.config.choiceRows.length === 0`
I get this error: `This condition will always return 'false' since the types '1' and '0' have no overlap`
So this syntax is for tuples.
And as we need arrays, `[string]` should be changed to `string[]` and `[[string]]` to `string[][]` 
😉

Reply:
Fixed